nova_vm/src/ecmascript/builtins/array.rs

```diff
@@ -80,6 +80,17 @@ impl Array {
         agent[self].elements.is_trivial(agent)
     }
 
+    pub(crate) fn shallow_clone(self, agent: &mut Agent) -> Array {
```

Comment:
**needs:** Please add a comment to mention that the array should be simple (no descriptors), and that otherwise this will do somewhat odd things (getter/setter indexes become holes). I know you have that comment on the elements side, but we should also have it here since this is the more likely "entry point" to call the function.